README.rst

```diff
@@ -89,18 +117,44 @@ This command will read the ``pyproject.toml``, install the dependencies of
 your plugin, and create a ``poetry.lock`` file to ensure everyone using your
 plugin gets the same version of dependencies. It is important that your
 dependencies are properly annotated with a version constraint (e.g.
-``manim:^0.1.1``, ``numpy:1.19.2``, etc...). Equally important to the
-dependencies specified here is that they do not conflict with manim's.
+``manim:^0.1.1``, ``numpy:*``, etc...). Equally important to the dependencies
+specified here is that they do not directly conflict with `Manim's
+<https://github.com/ManimCommunity/manim/blob/master/pyproject.toml>`_. If
+you want to update the dependencies specified in ``pyproject.toml``, use:
+
+.. code-block:: bash
+
+    poetry update
 
 See the official documentation for more information on `versioning
 <https://python-poetry.org/docs/dependency-specification/>`_ or the `install
 command <https://python-poetry.org/docs/cli/#install>`_. 
 
+
+Poetry allows for dependencies that are strictly for project developers.
+These are not installed by users. To add them to your project, update the
+``pyproject.toml`` file with the section followed by the dependencies:
+
+.. code::toml
+
+    [tool.poetry.dev-dependencies]
+    pytest = "*"
+    pylint = "*"
+
+The ``pytest`` package is a functional testing framework which you can use to
+run the test within the ``manim-YourPluginName/tests`` directory. You should
+create files which test the behavior and functionality of your plugin here.
+Test first development is a good practice to ensure your code behaves as
+intended before packaging and shipping your code publicly. Additionally, you
+can create Manimations that depend on your plugin which is another great way
+to ensure functionality.
+
 Uploading Your Project
 ----------------------
 
-By default, poetry is set to register the package/plugin to pypi. As soon as
-your plugin is useful locally, run the following:
+By default, poetry is set to register the package/plugin to PyPI. You'll need
+to register an account there to upload/update your plugin. As soon as your
+plugin is useful locally, run the following:
```

Comment:
Add a note about template repo and also, GitHub action to publish things. They have to set `PyPI` API token in secrets as `PYPI_API_TOKEN` and then just release of Github. Things will be pushed to PyPi automatically.

Reply:
I'm not sure if this is necessary to document, and I'm also unable to modify the GitHub Action (this requires repo edit permissions). I think you'll need to create project API Token on PyPI and the GitHub Secret so I can document this with a working example.

Reply:
> I think you'll need to create project API Token on PyPI and the GitHub Secret so I can document this with a working example.

I have done it when I have posted this comment. :) 

Just say, you can set you PyPi token in secrets tab, as  `PYPI_API_TOKEN `, and then do a release in Github, and things will be automatically pushed to PyPI.